include/deal.II/lac/petsc_vector_base.h

```diff
@@ -1003,10 +1020,37 @@ namespace PETScWrappers
 
       return *this;
     }
-  }
 
 
 
+    inline
+    PetscReal
+    VectorReference::real () const
+    {
+#ifndef PETSC_USE_COMPLEX
+      return static_cast<PetscScalar>(*this);
+#else
+      return PetscRealPart (static_cast<PetscScalar>(*this));
+#endif
+    }
+
+
+
+    inline
+    PetscReal
+    VectorReference::imag () const
+    {
+#ifndef PETSC_USE_COMPLEX
+      // This is a no op if complex numbers are not defined.
+      AssertThrow (false, ExcNotImplemented ());
+      return PetscReal (0);
```

Comment:
I would remove the assertion. If you're not using complex scalars, then their imaginary parts are simply zero.